### PR TITLE
Improve README.md

### PR DESCRIPTION
**Pull request by qurbonzoda**
The changes were developed within GSoC 2023 program by our mentee @wldeh. 
To facilitate review process: 
* The overhauled README focuses on setting up a Kotlin Multiplatform project for writing benchmarks and configuring their execution.
* There will be separate guides on setting up a Kotlin/JVM project or adding a benchmark module in an existing project.
* The non-existing documents referenced in this guide will be added in the followup PRs
* The README describes the next release (0.4.10) with https://github.com/Kotlin/kotlinx-benchmark/pull/163 , https://github.com/Kotlin/kotlinx-benchmark/pull/164 , https://github.com/Kotlin/kotlinx-benchmark/pull/165 , https://github.com/Kotlin/kotlinx-benchmark/pull/166 included

### CODE REVIEW

File: README.md
Comment (fzhinkin) [inline README.md:391]: `docs` are not here yet, so it is probably worth removing the reference and adding it back along with other documentation.

File: README.md
Comment (fzhinkin) [inline README.md:398]: The same issue with the link.

File: README.md
Comment (fzhinkin) [inline README.md:426]: And here, the link is being dangling as well.

File: README.md
Comment (fzhinkin) [inline README.md:431]: Not sure if we can merge it without either adding all new docs from `docs`, or leaving the old description.

**Review by fzhinkin — CHANGES_REQUESTED** [review]

File: README.md
Comment (qurbonzoda) [reply README.md:66]: As mentioned in the PR description, these changes describe the next release (0.4.10)
I will merge this PR after merging https://github.com/Kotlin/kotlinx-benchmark/pull/163 , https://github.com/Kotlin/kotlinx-benchmark/pull/164 , https://github.com/Kotlin/kotlinx-benchmark/pull/165 , https://github.com/Kotlin/kotlinx-benchmark/pull/166 and releasing a new version.

File: README.md
Comment (ilya-g) [inline README.md:334]: type argument is missing

File: README.md
Comment (ilya-g) [inline README.md:342]: Minor: `..<` instead of `until`

File: README.md
Comment (ilya-g) [inline README.md:9]: This badge (JS: IR supported) is likely obsolete.

File: README.md
Comment (Krobysin) [reply README.md:342]: clown

File: README.md
Comment (Krobysin) [reply README.md:334]: clown

File: README.md
Comment (qurbonzoda) [reply README.md]: Removed

File: README.md
Comment (qurbonzoda) [reply README.md]: Replaced

File: README.md
Comment (qurbonzoda) [reply README.md]: Added

File: README.md
Comment (qurbonzoda) [reply README.md:431]: This documentation is added in a separate PR: https://github.com/Kotlin/kotlinx-benchmark/pull/175

File: README.md
Comment (qurbonzoda) [reply README.md:426]: The document is added in a followup PR: https://github.com/Kotlin/kotlinx-benchmark/pull/168

File: README.md
Comment (qurbonzoda) [reply README.md:398]: The document is added in a followup PR: https://github.com/Kotlin/kotlinx-benchmark/pull/170

File: README.md
Comment (qurbonzoda) [reply README.md:389]: The document is added in a followup PR: https://github.com/Kotlin/kotlinx-benchmark/pull/184

**Review by fzhinkin — APPROVED** [review]

**Pull request merged**
